API/Controllers/BooksController.cs

```diff
@@ -0,0 +1,129 @@
+﻿using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using VL.Shared.Interfaces;
+using VL.Shared.Model;
+using static System.Reflection.Metadata.BlobBuilder;
+
+namespace API.Controllers
+{
+    [Authorize(AuthenticationSchemes = JwtBearerDefaults.AuthenticationScheme, Policy = "IsAllUsers")]
+    [Route("api/vl")]
+    [ApiController]
+    public class BooksController : ControllerBase
+    {
+        private readonly IBookService _bookService;
+
+        public BooksController(IBookService bookService)
+        {
+            _bookService = bookService;
+        }
+
+        /// <summary>
+        /// Get all books
+        /// </summary>
+        /// <returns>A list of books</returns>
+        [HttpGet]
+        [ProducesResponseType(typeof(bool), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status404NotFound)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status408RequestTimeout)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status500InternalServerError)]
```

Comment:
Yes many of these CAN happen but we only want to document what this controller is returning inside the actual BooksController


        [ProducesResponseType(typeof(bool), StatusCodes.Status200OK)]

Yes, but are you returning a bool?

        [ProducesResponseType(typeof(string), StatusCodes.Status400BadRequest)]

No because you aren't specifically returning a BadRequest in your Controller's implementation

        [ProducesResponseType(typeof(string), StatusCodes.Status401Unauthorized)]

Yes, because [Authorize...] is required to hit this Controller's route

        [ProducesResponseType(typeof(string), StatusCodes.Status403Forbidden)]

No, because 401 will be returned as only an authenticated user / token is required and you don't have a lower policy

        [ProducesResponseType(typeof(string), StatusCodes.Status404NotFound)]

No, right idea but a RESTful API should still return an empty array as a response (this will be more obvious when you do more full stack dev)

        [ProducesResponseType(typeof(string), StatusCodes.Status408RequestTimeout)]

No, sure a request can timeout but we're not expecting it here or implementing anything that would return it in the Controller's implementation

        [ProducesResponseType(typeof(string), StatusCodes.Status500InternalServerError)]

No, sure an internal server error can occur at any point in the request pipeline (even when decoding/evaluating your JWT for Authorize) but that's not anything that is specifically returned in your Controller's implementation